### PR TITLE
DEVDOCS-5646 [update] Checkouts, changes to discounts

### PR DESCRIPTION
**Pull request by bc-traciporter**
# [DEVDOCS-5646]

## What changed?
Updated checkout discount endpoint

## Release notes draft
V3/Discounts API can now be used to add item-level discount in addition to cart level discount

## Anything else?
Initial PR: [PR 1517](https://github.com/bigcommerce/api-specs/pull/1517)
[Checkout-7756](https://bigcommercecloud.atlassian.net/browse/CHECKOUT-7756)

ping @valentindellangela and @bc-charlesho for review

[DEVDOCS-5646]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-5646?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by valentindellangela — DISMISSED** [review]
Thank you

**Comment by valentindellangela** [comment]
@bc-charlesho Perhaps I can turn on the feature flag on Monday so when we release the dev doc, it's available?

**Comment by bc-charlesho** [comment]
@bc-traciporter can we commence this rollout next Monday Sydney time? When would you like us to roll out so that it is in sync with the publishing of the the dev docs?

File: reference/checkouts.v3.yml [hunk 1]
```diff
@@ -1604,14 +1604,19 @@ paths:
...
       description: |-
-        Adds a discount to an existing *checkout*.
-
-        This discount only applies to `line_items`. When you call this API, you clear out all existing discounts applied to line items, including product and order-based discounts.
+        Adds a discount to an existing checkout.
+        
+        Use this endpoint to apply the following discounts:
+        * Apply a manual discount to a cart. In this case, you can distribute the discount between each line item in the cart based on the item value.
+        * Apply a manual discount against a specific line item.
+        * Apply a manual discount at the item level with a cart-level discount.
```
Comment (bc-tgomez) [inline reference/checkouts.v3.yml:1612]:
```suggestion
        * Apply a manual discount at the item-level with a cart-level discount.
```

Comment (bc-tgomez) [reply reference/checkouts.v3.yml:1612]: Not sure if we want to add the hyphen in item-level?

File: reference/checkouts.v3.yml [hunk 2]
```diff
@@ -1604,14 +1604,19 @@ paths:
...
 
-        This endpoint splits the discount between line items based on the item value.
+        Notes:
+        * When you call this API, you clear out all existing discounts applied to line items, including product and order-based discounts.
+        * You cannot apply a manual discount to a specific line item if you have already applied a coupon or automatic promotion to it.
 
-        Required Fields
-        * discounted_amount
+        Required Fields:
+        * `discounted_amount` at the cart level or at the item level
```
Comment (bc-tgomez) [inline reference/checkouts.v3.yml:1619]:
```suggestion
        * `discounted_amount` at the cart-level or at the item level
```

Comment (bc-tgomez) [reply reference/checkouts.v3.yml:1619]: Not sure if we want to add the hyphen in item level here.

**Review by bc-tgomez — COMMENTED** [review]
I left one edit and a few comments. Not sure how you feel about *item-level since we use cart-level? Up to you! I'll approve right after.

**Review by bc-tgomez — APPROVED** [review]
Nice work!!

**Pull request merged**